### PR TITLE
CO-2815 Crm request migration v11

### PR DESCRIPTION
**Pull request by quentingigon**

### CODE REVIEW

File: crm_request/readme/CONTRIBUTORS.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+* Emanuel Cino <seicher@compassion.ch>
```
Comment (eicher31) [inline crm_request/readme/CONTRIBUTORS.rst]: Could you replace with Stephane Eicher?

**Review by eicher31 — CHANGES_REQUESTED** [review]

**Comment by eicher31** [comment]
Did you test the module with incoming mail server?

**Comment by quentingigon** [comment]
No I tested it with this technique: https://confluence.compassion.ch/pages/viewpage.action?pageId=25100297

File: crm_request/__manifest__.py [hunk 2]
```diff
@@ -10,20 +9,23 @@
...
         'python': ['detectlanguage']
     },
 
     # any module necessary for this one to work correctly
-    'depends': ['crm_claim_code',
-                'crm_claim_type',
+    'depends': ['crm_claim_code',                           # oca_addons/crm
+                'crm_claim_type',                           # oca_addons/crm
                 'mail',
                 'partner_contact_in_several_companies',
```
Comment (ecino) [inline crm_request/__manifest__.py]:
```suggestion
                'partner_contact_in_several_companies',  # oca_addons/partner-contact
```

**Review by ecino — CHANGES_REQUESTED** [review]
Repository `crm` should be added in `oca_dependencies.txt` to make Travis build.

**Comment by ecino** [comment]
@eicher31 Please review

**Comment by eicher31** [comment]
@quentingigon could you remove the file from crm_compassion please?

**Comment by ecino** [comment]
Remove the subject field adding the required attribute, it makes travis failing. (in file crm_request)

**Comment by eicher31** [comment]
The cards in the Kaban view are not clean.

**Comment by eicher31** [comment]
Could you remove the not concerned files?

File: crm_request/models/request_category.py [hunk 3]
```diff
@@ -38,3 +38,9 @@ def get_keys(self):
         for record in self:
             keywords_list.extend(record.keywords.split(','))
         return keywords_list
+
+    @api.model
+    def create(self, values):
+        values['template_id'] = \
+            self.env.ref("partner_communication.default_communication").id
```
Comment (eicher31) [inline crm_request/models/request_category.py]: You should use default value on the field in place

**Comment by eicher31** [comment]
The claim stage form has duplicate field.

File: crm_request/views/request_stage.xml [hunk 4]
```diff
@@ -1,10 +1,25 @@
...
+        <field name="name">crm.claim.stage.type.form</field>
+        <field name="model">crm.claim.stage</field>
+        <field name="inherit_id" ref="crm_claim.crm_claim_stage_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='sequence']" position="after">
+                <newline/>
+                <field name="case_default" />
+                <field name="claim_category"
+                     attrs="{'required':[('case_default','=',False)],
+                     'invisible':[('case_default','=',True)]}"/>
```
Comment (eicher31) [inline crm_request/views/request_stage.xml]: It doesn't work. The view has duplicate fields and this field here doesn't appear.

File: crm_request/models/request_stage.py [hunk 5]
```diff
@@ -9,3 +8,7 @@ class RequestStage(models.Model):
     _inherit = "crm.claim.stage"
 
     active = fields.Boolean(default=True)
+    claim_category = fields.Many2one(
```
Comment (eicher31) [inline crm_request/models/request_stage.py]: Why add this field in the object stage?

File: crm_request/views/request.xml [hunk 6]
```diff
@@ -98,7 +98,7 @@
                         <group>
                             <field name="team_id"/>
                             <field name="user_id" domain="[('share', '!=', True)]"/>
-                            <field name="claim_type" string="Type" required="True" options="{'no_create': True}"/>
+                            <field name="claim_category" string="Category" required="True" options="{'no_create': True}"/>
```
Comment (eicher31) [inline crm_request/views/request.xml]: You can remove the required attribute.

File: crm_request/data/crm_request_data.xml [hunk 7]
```diff
@@ -4,13 +4,11 @@
...
         <field name="case_default" eval="True"/>
-        <field name="claim_common" eval="True"/>
     </record>
 
     <record model="crm.claim.stage" id="stage_wait_support">
         <field name="name">Waiting for support</field>
         <field name="sequence">15</field>
         <field name="case_default" eval="True"/>
-        <field name="claim_common" eval="True"/>
     </record>
```
Comment (eicher31) [inline crm_request/data/crm_request_data.xml:15]: Create a category "Undefined" and use it as default in the request.

File: crm_request/models/request.py [hunk 8]
```diff
@@ -19,12 +18,12 @@ class CrmClaim(models.Model):
...
     date = fields.Datetime(string='Date', readonly=True, index=False)
     name = fields.Char(compute='_compute_name', store=True)
-    subject = fields.Char(required=True)
+    subject = fields.Char()
     alias_id = fields.Many2one(
         'mail.alias', 'Alias',
         help="The destination email address that the contacts used.")
     code = fields.Char(string='Number')
-    claim_type = fields.Many2one(string='Type')
+    claim_category = fields.Many2one("crm.claim.category", string='Category')
```
Comment (eicher31) [inline crm_request/models/request.py]: Use default category

**Review by eicher31 — CHANGES_REQUESTED** [review]

**Comment by eicher31** [comment]
Create script migration to move record from crm_request_type to crm_request_category

File: crm_request/models/request_stage.py (hunk 5 above)
Comment (quentingigon) [reply crm_request/models/request_stage.py]: I removed it

File: crm_request/security/ir.model.access.csv [hunk 9]
```diff
@@ -1,2 +1,3 @@
 id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
 admin_access_holiday_closure,admin_access_holiday_closure,model_holiday_closure,sales_team.group_sale_manager,1,1,1,1
+access_crm_stages,access_crm_stages,model_crm_claim_stage,sales_team.group_sale_manager,1,1,1,1
```
Comment (eicher31) [inline crm_request/security/ir.model.access.csv:3]: Add permission to the model crm_claim_categorie, so only manager can create it.

File: crm_request/models/request.py [hunk 10]
```diff
@@ -19,12 +18,14 @@ class CrmClaim(models.Model):
...
     date = fields.Datetime(string='Date', readonly=True, index=False)
     name = fields.Char(compute='_compute_name', store=True)
-    subject = fields.Char(required=True)
+    subject = fields.Char()
     alias_id = fields.Many2one(
         'mail.alias', 'Alias',
         help="The destination email address that the contacts used.")
     code = fields.Char(string='Number')
-    claim_type = fields.Many2one(string='Type')
+    claim_category = fields.Many2one(
```
Comment (eicher31) [inline crm_request/models/request.py:26]: use only category

File: crm_request/views/request.xml [hunk 11]
```diff
@@ -98,7 +98,7 @@
                         <group>
                             <field name="team_id"/>
                             <field name="user_id" domain="[('share', '!=', True)]"/>
-                            <field name="claim_type" string="Type" required="True" options="{'no_create': True}"/>
+                            <field name="claim_category" string="Category" options="{'no_create': True}"/>
```
Comment (eicher31) [inline crm_request/views/request.xml:101]: Rename category

**Review by eicher31 — CHANGES_REQUESTED** [review]

File: crm_request/models/request.py (hunk 10 above)
Comment (quentingigon) [reply crm_request/models/request.py:26]: bad idea

File: crm_request/views/request.xml (hunk 11 above)
Comment (quentingigon) [reply crm_request/views/request.xml:101]: idea

File: crm_request/models/request.py [hunk 12]
```diff
@@ -19,12 +18,14 @@ class CrmClaim(models.Model):
...
+        "crm.claim.category", string='Category',
+        default='self.env.ref("crm_request.stage_undefined")')
```
Comment (ecino) [inline crm_request/models/request.py]: The default expects a string. If you want to interpret it you should use lambda function:
```suggestion
        default=lambda self: self.env.ref("crm_request.stage_undefined"))
```

Comment (quentingigon) [reply crm_request/models/request.py]: I corrected it using a compute, before seeing your message. Is it ok ?

**Comment by ecino** [comment]
Too bad, one pep8 issue : `crm_request/migrations/11.0.0.1.0/post-migration.py:22:43: W291 trailing whitespace`

**Pull request merged**